cmd/jaeger/internal/extension/jaegerstorage/factoryadapter/writer_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package factoryadapter
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/memory"
+	spanstoreMocks "github.com/jaegertracing/jaeger/storage/spanstore/mocks"
+)
+
+func TestWriteTraces(t *testing.T) {
+	memstore := memory.NewStore()
+	traceWriter := &TraceWriter{
+		spanWriter: memstore,
+	}
+
+	err := traceWriter.WriteTraces(context.Background(), makeTraces())
+	require.NoError(t, err)
```

Comment:
nit: test should validate that the data reached the storage. If WriteTraces() was noop it would pass the current test.